torch/csrc/jit/codegen/cuda/transform_iter.cpp

```diff
@@ -440,6 +473,277 @@ int BestEffortReplay::findFirstMismatchedID(
   return td1->nDims();
 }
 
+namespace {
+
+// Maps that track information relevant to best effort replay about broadcast
+// axes in consumer that are not in producer
+//
+// For example if we have consumer: T0[i0, b1, b2, i3] and producer:
+// T1[i0, i3]
+//
+// If consumer transformations are:
+// -> T[i0, b1o, b1i, b2o, b2i, i3]
+// -> T[i0*b1i*b2o, b1o, b2i, i3]
+// -> T[i0*b1i*b2o*i3, b1o, b2i]
+//
+// forwarding_map would forward i0->i0*b1i*b2o
+// compliment_map would have the entry i0->b1o and i0*b1i->b2i
```

Comment:
Copying as github says the comment was outdated:

Yes, I understand what this class is designed to do now, but it was a little confusing as the comment wasn't exactly matching with my understanding of the logic, so I was unsure whether I was reading the code wrong or there was something wrong in the comment.

I'd suggest to illustrate two things by the example: 1) the end result we need and 2) what this particular class does for that. In the example case, what we eventually want to do is to detect b1o and b2i as leaves. For that, ConsumerForwardingInfo creates mappings of i0->b1i and i0*b1i->b2o.